### PR TITLE
fix: retry BulkWriter deletes that fail with RST_STREAM error

### PR DESCRIPTION
**Pull request by thebrianchen**

### CODE REVIEW

File: dev/src/bulk-writer.ts [hunk 1]
```diff
@@ -316,10 +316,14 @@ export class BulkWriter {
    * @private
    */
   private _errorFn: (error: BulkWriterError) => boolean = error => {
+    const isRstStreamError =
+      (error.code as number) === Status.INTERNAL &&
```
Comment (schmidt-sebastian) [inline dev/src/bulk-writer.ts:320]: 1) We should be able to retry Status.INTERNAL regardless of whether this is a RST_STREAM error.
2) We should only add this retry for deletes.

File: dev/src/bulk-writer.ts [hunk 2]
```diff
@@ -316,10 +316,13 @@ export class BulkWriter {
    * @private
    */
   private _errorFn: (error: BulkWriterError) => boolean = error => {
+    const isRetryableDeleteError =
+      error.operationType === 'delete' &&
+      (error.code as number) === Status.INTERNAL;
     const retryCodes = getRetryCodes('batchWrite');
     return (
       error.code !== undefined &&
```
Comment (schmidt-sebastian) [inline dev/src/bulk-writer.ts:324]: Do we actually need this undefined check? Would retryCodes.include() just return false?

**Review by schmidt-sebastian — APPROVED** [review]

File: dev/src/bulk-writer.ts (hunk 2 above)
Comment (thebrianchen) [reply dev/src/bulk-writer.ts]: nope! removed.

File: dev/src/bulk-writer.ts (hunk 1 above)
Comment (jyane) [reply dev/src/bulk-writer.ts]: Would you please explain why the retry should be only for deletes?
The RST could happen on any other methods, this limitation comes from idempotency?

For example, Spanner automatically retries transaction commits, regardless of the queries within the txn. I wonder if there is a firestore specific thing here.

**Pull request merged**